### DISCUSSION TITLE
Streaming response from CH to Grafana

### DISCUSSION THREAD

**Discussion by Alexul** [grafana/clickhouse-datasource#866] [Ideas]
Problem
When visualizing data from CH in Grafana (for example time-series graph with count per 1 minute) plugin waits till CH responds with complete answer. On big timeframes response time from CH can reach minutes, making the interface/user experience slow/bad

Proposal
Use pagination to stream response from CH to Grafana in click-house plugin. Graph should start loading in chunks, starting from now() resulting in better user experience. Especially true in scenarios with different storage tiers

Additional info
Not clear how to not break alerting queries with that feature. Maybe we need to ask Grafana to add a clear differentiation between alert and user query. That can also be useful to define different users/endpoints in CH grafana plugin for alerts/user queries (typically alerting user need to have more strict query limits)

WDYT?

**Comment by bossinc** [comment]
@SpencerTorres Prometheus DS has [Incremental dashboard queries](https://grafana.com/docs/grafana/latest/datasources/prometheus/#incremental-dashboard-queries-beta).  I am trying to find out more information about this feature.

**Comment by SpencerTorres** [comment]
see https://github.com/grafana/clickhouse-datasource/discussions/404#discussioncomment-10347581
